File: src/main/java/com/jcloisterzone/event/TokenRemovedEvent.java
package com.jcloisterzone.event;

import com.jcloisterzone.board.pointer.BoardPointer;
import com.jcloisterzone.game.Token;

public class TokenRemovedEvent extends PlayEvent {

    private static final long serialVersionUID = 1L;

    private final Token token;
    private final BoardPointer pointer;
    private final int count;
    private final boolean forced;

    public TokenRemovedEvent(PlayEventMeta metadata, Token token, BoardPointer pointer, int count, boolean forced) {
        super(metadata);
        this.token = token;
        this.pointer = pointer;
        this.count = count;
        this.forced = forced;
    }

    public Token getToken() {
        return token;
    }

    public BoardPointer getPointer() {
        return pointer;
    }

    public int getCount() {
        return count;
    }

    public boolean getForced() {
        return forced;
    }
}

File: src/main/ts/com/jcloisterzone/event/TokenRemovedEvent.ts
import type { BoardPointer } from "../board/pointer/BoardPointer.js";
import type { Token } from "../game/Token.js";
import { PlayEvent, type PlayEventMeta } from "./PlayEvent.js";

/** A supply token was removed from the board (e.g. a meteorite destroys a gold piece). */
export class TokenRemovedEvent extends PlayEvent {
  static readonly simpleName = "TokenRemovedEvent";

  constructor(
    metadata: PlayEventMeta,
    private readonly token: Token,
    private readonly pointer: BoardPointer,
    private readonly count: number,
    private readonly forced: boolean,
  ) {
    super(metadata);
  }

  getToken(): Token {
    return this.token;
  }
  getPointer(): BoardPointer {
    return this.pointer;
  }
  getCount(): number {
    return this.count;
  }
  getForced(): boolean {
    return this.forced;
  }
}
